astropy/io/fits/diff.py

```diff
@@ -1236,10 +1244,21 @@ def report_diff_values(fileobj, a, b, ind=0):
     for line in difflib.ndiff(str(a).splitlines(), str(b).splitlines()):
         if line[0] == '-':
             line = 'a>' + line[1:]
+            if typea != typeb:
+                padding = max(len(typea.__name__), len(typeb.__name__)) + 3
+                typename = '(' + typea.__name__ + ') '
+                line = typename.rjust(padding) + line
```

Comment:
Actually, just one more suggestion--`padding` can be assigned higher up, outside the loop, since it will always be the same and doesn't need to be computed more than once.